### ISSUE TITLE
RestEndpoint.extend with no-argument fetch doesn't work with useFetcher

### ISSUE THREAD

**Issue by yunyu** [coinbase/rest-hooks#1474]
**React version** 17.0.2

**Concurrent mode** no

**Rest Hooks version** 6.1.7

**Describe the bug**
Returning RestEndpoint.extend results in a type error when trying to use it in useFetcher. It works fine with an object spread. This is an example from the documentation: https://resthooks.io/docs/guides/extending-endpoints#custom-get

**To Reproduce**
This doesn't work:

```ts
export class ProfileResource extends Resource {
  static urlRoot = "profile";

  readonly id!: string;
  readonly email!: string;

  pk() {
    return this.id;
  }

  static current<T extends typeof Resource>(this: T) {
    const endpoint = this.detail();
    return endpoint.extend({
      fetch() {
        return endpoint(this);
      },
      url() {
        return ProfileResource.urlRoot;
      },
    });
  }
}

// Elsewhere in application
const fetchCurrentProfile = useFetcher(ProfileResource.current());
```

![image](https://user-images.githubusercontent.com/8008350/140651286-23e5a447-9659-4711-9ab8-84c90aae0c4b.png)

**Expected behavior**
useFetcher should be able to take in outputs from RestEndpoint.extend

**Additional context**
Object spread works and doesn't result in a type error:
```ts
  static current<T extends typeof Resource>(this: T) {
    const endpoint = this.detail();
    return {
      ...endpoint,
      fetch() {
        return endpoint(this);
      },
      url() {
        return ProfileResource.urlRoot;
      },
    };
  }
```

**Comment by yunyu** [comment]
Unrelated issue: useFetcher doesn't seem to allow passing in any arguments for custom endpoints:

![image](https://user-images.githubusercontent.com/8008350/140651445-6a41ab2b-a5f8-4606-9461-c7a7e9864f5e.png)

**Comment by ntucker** [comment]
Workaround is to put an argument in your fetch call and just ignore it. I can losen the types for useFetcher tho. If you're using v6 you should try using controller.fetch from useController().

**Comment by ntucker** [comment]
PS) Due to a weird TypeScript bug you'll also want to **Either**

1) Add `schema: this`
2) Explicitly type the return value

More info here: https://resthooks.io/docs/guides/rest-types#a-limitation

**Comment by ntucker** [comment]
Fixed in [@rest-hooks/endpoint@2.1.0](https://www.npmjs.com/package/@rest-hooks/endpoint)
